src/viam/app/location/client.py

```diff
@@ -0,0 +1,720 @@
+import json
+from datetime import datetime
+from typing import Any, List, Mapping, Optional
+
+from grpclib.client import Channel
+from typing_extensions import Self
+
+from viam import logging
+from viam.proto.app import (
+    AppServiceStub,
+    CreateLocationRequest,
+    CreateLocationResponse,
+    DeleteLocationRequest,
+    DeleteLocationResponse,
+    DeleteRobotPartRequest,
+    DeleteRobotPartResponse,
+    DeleteRobotRequest,
+    DeleteRobotResponse,
+)
+from viam.proto.app import Fragment as FragmentPB
+from viam.proto.app import (
+    GetFragmentRequest,
+    GetFragmentResponse,
+    GetLocationRequest,
+    GetLocationResponse,
+    GetRobotPartLogsRequest,
+    GetRobotPartLogsResponse,
+    GetRobotPartRequest,
+    GetRobotPartResponse,
+    GetRobotPartsRequest,
+    GetRobotPartsResponse,
+    GetRobotRequest,
+    GetRobotResponse,
+    ListFragmentsRequest,
+    ListFragmentsResponse,
+    ListLocationsRequest,
+    ListLocationsResponse,
+    ListRobotsRequest,
+    ListRobotsResponse,
+    Location,
+)
+from viam.proto.app import LogEntry as LogEntryPB
+from viam.proto.app import (
+    MarkPartForRestartRequest,
+    MarkPartForRestartResponse,
+    NewRobotPartRequest,
+    NewRobotPartResponse,
+    NewRobotRequest,
+    NewRobotResponse,
+    Robot,
+)
+from viam.proto.app import RobotPart as RobotPartPB
+from viam.proto.app import (
+    SharedSecret,
+    ShareLocationRequest,
+    ShareLocationResponse,
+    UnshareLocationRequest,
+    UnshareLocationResponse,
+    UpdateLocationRequest,
+    UpdateLocationResponse,
+    UpdateRobotPartRequest,
+    UpdateRobotPartResponse,
+    UpdateRobotRequest,
+    UpdateRobotResponse,
+)
+from viam.utils import dict_to_struct, struct_to_dict
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LocationClient:
+    """gRPC client for method calls to app.
+
+    Constructor is used by `AppClient` to instantiate relevant service stub. Calls to `LocationClient`  methods should be made through
+    `AppClient`.
+    """
+
+    class RobotPart:
```

Comment:
All of these class should live OUTSIDE of `LocationClient`, since they could be used anywhere/everywhere